setup.py

```diff
@@ -55,8 +52,10 @@
             'pytest',
             'pytest-cov',
             'nbval',
+            'numpy>=1.7',
             'scipy>=0.19.0',
             # examples/tutorials
+            'observations>=0.1.4',
```

Comment:
I believe we should also include `observations` in `notebooks` since some examples and tutorials depend on them? Or maybe we should keep it in `install_requires`? I'm not sure whether to cater to new users for whom it would be easier to include `observations` in `install_requires`, or to power users who might want to deploy Pyro with minimal dependencies. @neerajprad do you have an opinion?

Reply:
I think we should just add it to `notebooks` to keep core Pyro requirements separate (and small), and not have to add in ad-hoc libraries that might be needed by some example. Even `matplotlib` is required for running some of the notebooks and is not listed as a `notebook` requirement, which we should likewise add.

Reply:
I failed to find a good way to include `extras` into `test`, so I make a new variable EXTRAS_REQUIRE. We can keep the old way but have to modify travis config (like `pip install -e [extras,test]`).